src/net/rpc.rs

```diff
@@ -1509,6 +1510,86 @@ impl ConversationHttp {
         return response.send(http, fd).map(|_| ());
     }
 
+    /// Handle a GET health.
+    /// If the node's height is not up to the max height of all of its bootstrap nodes, we return
+    /// false for the health check.
+    /// Additionally return what percent of the max height the node height is at.
+    fn handle_get_health<W: Write>(
+        http: &mut StacksHttp,
+        fd: &mut W,
+        req: &HttpRequestType,
+        network_id: u32,
+        sortdb: &SortitionDB,
+        chainstate: &StacksChainState,
+        peerdb: &PeerDB,
+        block_stats: Option<&HashMap<NeighborKey, NeighborBlockStats>>,
+    ) -> Result<(), net_error> {
+        let response_metadata = HttpResponseMetadata::from(req);
+
+        let block_height = match chainstate.get_stacks_chain_tip(sortdb)? {
+            Some(tip) => tip.height,
+            None => 0,
+        };
+
+        let initial_neighbors =
+            PeerDB::get_valid_initial_neighbors(peerdb.conn(), network_id, block_height).unwrap();
+
+        if initial_neighbors.len() == 0 {
+            // no bootstrap nodes found, unable to determine health.
+            let response = HttpResponseType::NotFound(
+                response_metadata.clone(),
+                format!("No viable bootstrap peers found, unable to determine health"),
+            );
+            return response.send(http, fd).map(|_| ());
+        }
+
+        // get max block height amongst bootstrap nodes
+        let response = if let Some(block_stats) = block_stats {
+            let mut max_height: u64 = 1;
+            let mut stats_obtained = false;
+            for neighbor in initial_neighbors.iter() {
+                let nk = &neighbor.addr;
+                match block_stats.get(nk) {
+                    Some(neighbor_block_stats) => {
+                        if neighbor_block_stats.status == NodeStatus::Online {
+                            let height = neighbor_block_stats.inv.get_block_height();
+                            max_height = max_height.max(height);
+                            stats_obtained = true;
+                        }
+                    }
+                    None => {}
+                }
+            }
+
+            if !stats_obtained {
+                let response = HttpResponseType::NotFound(
+                    response_metadata.clone(),
+                    "Couldn't obtain stats on any bootstrap peers, unable to determine health"
+                        .to_string(),
+                );
+                return response.send(http, fd).map(|_| ());
+            }
+
+            let data = if block_height >= (max_height - 1) {
+                GetHealthResponse {
+                    is_healthy: true,
```

Comment:
Rather than "is_healthy", I wonder if you could call this something like "matches_peers" or something.

That way, we can eventually add different statistics here, and then the client can determine whether they think the node "is health" based on the other signals, like "matches_peers".